Common/TimeKeeper.cs

```diff
@@ -28,7 +28,7 @@ public class TimeKeeper : ITimeKeeper
     {
         private DateTime _utcDateTime;
 
-        private readonly Dictionary<DateTimeZone, LocalTimeKeeper> _localTimeKeepers;
+        private readonly Dictionary<string, LocalTimeKeeper> _localTimeKeepers;
```

Comment:
cool! out of curiosity, what was `DateTimeZone` using for it's `GetHashCode` and `Equals` functions? Maybe it was comparing some underlying definitional object?